src/components/CustomFileInput/index.js

```diff
@@ -36,10 +36,39 @@ export default class CustomFileInput extends Component<Props> {
     event.currentTarget.value = '' // Allow re-uplading the same file
   }
 
+  handleDragOverEvent = (ev: DragEvent) => {
+    // Prevent default behavior (Prevent file from being opened)
+    ev.preventDefault();
+  }
+
+  getFileFromDragEvent = (ev: SyntheticDragEvent<HTMLSpanElement>): ?File => {
+    const dataTransfer: DataTransfer = ev.dataTransfer;
+    const items: DataTransferItemList = dataTransfer.items;
+    const files: FileList = dataTransfer.files;
+
+    if (items) {
+      const file: ?File = items[0].getAsFile();
+      return file;
+    } else if (files) {
+      const file: ?File = files[0];
+      return file;
+    }
+  }
+
+  handleDropEvent = (ev: SyntheticDragEvent<HTMLSpanElement>) => {
+    // Prevent default behavior (Prevent file from being opened)
+    ev.preventDefault();
+
+    const file: ?File = this.getFileFromDragEvent(ev);
+    if (file) {
```

Comment:
Minor: this can also be done as a one-liner `file && this.props.onChange(file);`

Reply:
will change